### PR TITLE
more ben: Add nwc profile to vss

### PR DESCRIPTION
**Pull request by bennyhodl**
Saves created NWC profiles to vss.

Version is updated on creating new NWC profiles in `indexed_db`. Could use some direction on checking the vss storage.

### CODE REVIEW

**Comment by TonyGiorgio** [comment]
Nice. We should eventually have more things in VSS but kept it to critical data. However I can see NWC being annoying to keep adding if someone reset their wallet. We want to change to a more deterministic key generation approach for NWC based on the profile name, but haven't gotten around to that yet. We may have to do that first before this, since doing so will break all NWCs.

Quick code base looks alright, I'll have to let @benthecarman comment on the specifics since he has more experience with these two things.

**Comment by bennyhodl** [comment]
@benthecarman rebased and squashed to one commit

File: mutiny-core/src/nostr/mod.rs [hunk 1]
```diff
@@ -60,6 +78,8 @@ pub struct NostrManager<S: MutinyStorage> {
     pub storage: S,
     /// Lock for pending nwc invoices
     pending_nwc_lock: Arc<Mutex<()>>,
+    // Versioned nwc profiles for VSS. Option for backwards compatability.
+    nwc_profile_version: Option<NwcProfileVersion>,
```
Comment (benthecarman) [inline mutiny-core/src/nostr/mod.rs:82]: When this is `None` it is never set to be `Some` after it is updated. Probably what would be best is to not have this be optional and on init, if we don't have it, create a new one with version=0

File: mutiny-core/src/nostr/mod.rs [hunk 2]
```diff
@@ -568,12 +608,15 @@ impl<S: MutinyStorage> NostrManager<S> {
             .map(|profile| NostrWalletConnect::new(&context, xprivkey, profile).unwrap())
             .collect();
 
+        let nwc_profile_version = NwcProfileVersion::new();
```
Comment (benthecarman) [inline mutiny-core/src/nostr/mod.rs:611]: this needs to be read from storage, otherwise you'll always be starting with version 0

**Review by benthecarman — CHANGES_REQUESTED** [review]
This still also doesn't handle backwards compat for reading the nwc configs from _before_ this PR. 

Also looks like there are some more places where we write the nwc storage without doing vss

**Pull request closed without merging**